### PR TITLE
ARROW-10364: [Dev][Archery] Add support for semver 2.13.0

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kszucs** [comment]
I didn't see it before applied https://github.com/apache/arrow/pull/8506 which also fixes the issue.

**Comment by kou** [comment]
Which approach do you like?
Should we close this?

File: dev/archery/archery/release.py [hunk 1]
```diff
@@ -39,16 +39,19 @@ class Version(SemVer):
 
     __slots__ = SemVer.__slots__ + ('released', 'release_date')
```
Comment (terencehonles) [inline dev/archery/archery/release.py:40]: Aside: slots shouldn't need to be duplicated like this

**Review by terencehonles — COMMENTED** [review]
> Which approach do you like?
> Should we close this?

I'll let you both work it out, but just calling out that they should work in tandem and this would fix any potential future string comparisons that are made that aren't covered with the test suite.

**Comment by emkornfield** [comment]
@kou @kszucs any resolution on whether we want to merge this?

File: dev/archery/archery/release.py (hunk 1 above)
Comment (kou) [reply dev/archery/archery/release.py]: Thanks.
I've removed it.

**Comment by kou** [comment]
@kszucs Can I merge this?

**Comment by pitrou** [comment]
Ping @kszucs

**Comment by pitrou** [comment]
Merging, we'll see if anyone complains :-)

**Pull request closed without merging**